zerver/lib/upload.py

```diff
@@ -602,16 +604,16 @@ def get_emoji_url(self, emoji_file_name: str, realm_id: int) -> str:
         bucket = settings.S3_AVATAR_BUCKET
         emoji_path = RealmEmoji.PATH_ID_TEMPLATE.format(realm_id=realm_id,
                                                         emoji_file_name=emoji_file_name)
-        return "https://%s.%s/%s" % (bucket, self.connection.DefaultHost, emoji_path)
+        return "https://%s.%s/%s" % (bucket, self.session.DefaultHost, emoji_path)
 
     def upload_export_tarball(self, realm: Optional[Realm], tarball_path: str) -> str:
```

Comment:
This is the remaining change which is not covered by #10370, which I am not sure what to be done. @andersk I think it is safer if you take over this boto3 upgrade.